UN-OCHA/unocha-site
# [UNO-685] Contact form

Refs: UNO-685

This adds the global contact form using the built-in contact module and a new taxonomy vocabulary ("contacts") to manage the list of "categories" selectable on the form. 

Each "contact" term has a label ("category") and a list of email addresses. Editors can edit those terms.

In addition there is a `unocha_contact` module to alter the email and use the emails of the selected category.

### Deployment

After deploying, the follow should be done:

1. Delete the `/node/59` node
2. Add a URL alias to the contact form `/admin/config/search/path/add`:

<img width="779" alt="Screenshot 2023-07-07 at 9 46 28" src="https://github.com/UN-OCHA/unocha-site/assets/696348/bdcc947b-383b-41a0-a6f0-fa2f04faa40e">

3. Add a menu link entry for the contact form `/admin/structure/menu/manage/main/add`:

<img width="699" alt="Screenshot 2023-07-07 at 9 48 08" src="https://github.com/UN-OCHA/unocha-site/assets/696348/67e427b2-0020-4386-b0b1-fc5e1e7da222">

4. Add contact terms as per the spreadsheet in the ticket. Each category can have several emails (comma separated list in the spreadsheet email column):

<img width="685" alt="Screenshot 2023-07-07 at 9 52 07" src="https://github.com/UN-OCHA/unocha-site/assets/696348/cd7a17e2-e045-4690-b4fe-42bb8c71ad85">

### Tests

**Mailhog setup**

To be able to test locally, you will problably need a mailhog instance to see the result of the sent emails, otherwise deploy this feature branch on `dev` and test there.

You can try adding the following to the `docker-compose.yml` under `services:` for the site stack:

```yml
  mailhog:
    image: mailhog/mailhog:v1.0.1
    hostname: unocha-local-mailhog
    container_name: unocha-local-mailhog
    user: root
    entrypoint:
      - MailHog
    environment:
      ENVIRONMENT: dev
      NGINX_SERVERNAME: mh.unocha-local.test
      VIRTUAL_HOST: mh.unocha-local.test
      VIRTUAL_PORT: 8025
      VIRTUAL_NETWORK: ${PROXY:-proxy}
      HTTPS_METHOD: noredirect
      MH_SMTP_BIND_ADDR: 0.0.0.0:25
    logging:
      driver: none
    labels:
      caddy: mh.unocha-local.test
      caddy.reverse_proxy: "{{upstreams}}"
    expose:
      - "25"
    networks:
      default:
        aliases:
          - smtp
      proxy:
        aliases:
          - smtp.unocha-local
```

Possibly add an entry for `mh.unocha-local.test` in your `/etc/hosts`.

**Testing**

1. Checkout the branch, clear the cache, import the config
2. Follow the steps from the `Deployment notes` above
3. Go to `/contact-us` as anonymous
4. Fill in the form and click send
5. Check that the email appears in mailhog, it should resemble the following:

<img width="1195" alt="Screenshot 2023-07-07 at 10 00 19" src="https://github.com/UN-OCHA/unocha-site/assets/696348/8ede420d-fd81-44ea-ac2c-ac61b0bbefa3">

6. Repeat as a  logged in user
